src/nvim/ex_cmds2.c

```diff
@@ -2719,16 +2719,13 @@ static char_u *get_str_line(int c, void *cookie, int indent, bool do_concat)
   while (!(p->buf[i] == '\n' || p->buf[i] == '\0')) {
     i++;
   }
-  char buf[2046];
-  char *dst;
-  dst = xstpncpy(buf, (char *)p->buf + p->offset, i - p->offset);
-  if ((uint32_t)(dst - buf) != i - p->offset) {
-    smsg(_(":source error parsing command %s"), p->buf);
-    return NULL;
-  }
-  buf[i - p->offset] = '\0';
+  size_t line_length = i - p->offset;
+  garray_T ga;
+  ga_init(&ga, (int)sizeof(char_u), (int)line_length);
+  ga_concat_len(&ga, (char *)p->buf + p->offset, line_length);
+  ga_append(&ga, '\0');
```

Comment:
Was there a special reason to use an `garray_T` here? Otherwise it looks like we could replace the use of the `garray_T` object with   `char_u *buf = xmemdupz(p->buf + p->offset, line_length);` and later return `buf`.

Reply:
> Was there a special reason to use an `garray_T` here?

No, there was no particular reason. The only compelling reason was my unfamiliarity with alternative methods. 😅 

>  Otherwise it looks like we could replace the use of the `garray_T` object with `char_u *buf = xmemdupz(p->buf + p->offset, line_length);` and later return `buf`.

The only extra thing this function does, is replacing `\n` with `\0`/ adding a `\0` termination  essentially .

Reply:
When I suggested using `garray_T`, I thought there might be more iteration needed (like in the source from file case), but you're right that we know the size up front.  `xmemdupz()` is a better choice.